pkg/logql/range_vector_test.go

```diff
@@ -56,6 +56,37 @@ func newPoint(t time.Time, v float64) promql.Point {
 	return promql.Point{T: t.UnixNano() / 1e+6, V: v}
 }
 
+func Benchmark_RangeVectorIterator(b *testing.B) {
+	b.ReportAllocs()
+	tt := struct {
+		selRange   int64
+		step       int64
+		offset     int64
+		start, end time.Time
+	}{
+		(5 * time.Second).Nanoseconds(), // no overlap
+		(30 * time.Second).Nanoseconds(),
+		0,
+		time.Unix(10, 0),
+		time.Unix(100, 0),
+	}
+
+	for i := 0; i < b.N; i++ {
+		i := 0
+		it, err := newRangeVectorIterator(newfakePeekingSampleIterator(),
+			&syntax.RangeAggregationExpr{Operation: "count_over_time"}, tt.selRange,
```

Comment:
nit: `s/count_over_time/syntax.OpRangeTypeCount/`

Just to be more static check friendly.

```suggestion
			&syntax.RangeAggregationExpr{Operation: syntax.OpRangeTypeCount}, tt.selRange,
```